src/mito2/src/sst/parquet/reader.rs

```diff
@@ -159,18 +177,24 @@ impl ParquetReaderBuilder {
         let parquet_meta = self.read_parquet_metadata(&file_path, file_size).await?;
         // Decodes region metadata.
         let key_value_meta = parquet_meta.file_metadata().key_value_metadata();
-        let region_meta = Self::get_region_metadata(&file_path, key_value_meta)?;
-        let read_format = ReadFormat::new(Arc::new(region_meta));
+        // Gets the metadata stored in the SST.
+        let region_meta = Arc::new(Self::get_region_metadata(&file_path, key_value_meta)?);
+        // Lists all column ids to read, we always use the expected metadata if possible.
+        let column_ids = self.projection.clone().unwrap_or_else(|| {
+            let expected_meta = self.expected_metadata.as_ref().unwrap_or(&region_meta);
+            expected_meta
+                .column_metadatas
+                .iter()
+                .map(|col| col.column_id)
+                .collect()
+        });
+        let read_format = ReadFormat::new(region_meta.clone(), &column_ids);
```

Comment:
Hopefully we can eliminate a clone
```suggestion
        let read_format = if let Some(column_ids) = &self.projection {
            ReadFormat::new(region_meta.clone(), column_ids)
        } else {
            let expected_meta = self.expected_metadata.as_ref().unwrap_or(&region_meta);
            let column_ids = expected_meta
                .column_metadatas
                .iter()
                .map(|col| col.column_id)
                .collect::<Vec<_>>();
            ReadFormat::new(region_meta.clone(), &column_ids)
        };
```